cmd/osbuild-composer/composer.go

```diff
@@ -75,10 +75,17 @@ func NewComposer(config *ComposerConfigFile, stateDir, cacheDir string, logger *
 
 func (c *Composer) InitWeldr(repoPaths []string, weldrListener net.Listener) error {
 	archName := common.CurrentArch()
+	fallback := false
+	var fallback_name string
+	var fallback_repos map[string][]rpmmd.RepoConfig
 
 	hostDistro := c.distros.FromHost()
 	if hostDistro == nil {
-		return fmt.Errorf("host distro is not supported")
+		hostDistro, fallback_name, fallback_repos = c.distros.FallbackHost()
+		fallback = true
+		if hostDistro == nil {
+			return fmt.Errorf("host distro is not supported")
+		}
```

Comment:
Personally I would prefer to move this logic to the `FromHost()` method and have it as the only exported method. The `FallbackHost()` could be called inside of the `FromHost()`. So in the end if the `FromHost()` returns a `distro.Distro`, it is either the supported or unsupported/fallback distro or `nil` and the distro is not supported at all.